File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/AlumnoService.java
package com.taekwondo.service;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.taekwondo.model.Alumno;

public interface AlumnoService {
	public abstract List<Alumno> getAlumnos();
	public abstract ResponseEntity<Object> getAlumno(int id);
	public abstract ResponseEntity<Object> createAlumno(Alumno alumno);
	public abstract void updateAlumno(int id, Alumno alumno);
	public abstract void deleteAlumno(int id);
}

File: proyectoTaeKwonDo/src/app/_services/alumno.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Alumno } from '../_models/alumno';

@Injectable({
  providedIn: 'root'
})
export class AlumnoService {

  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getAlumnos(){
    return this.http.get(this.API_URI+'/alumno');
  }

  getAlumno(id: number){
    return this.http.get(this.API_URI+'/alumno/'+id);
  }

  createAlumno(alumno: Alumno){
    return this.http.post(this.API_URI+'/alumno', alumno);
  }

  updateAlumno(alumno: Alumno){
    return this.http.put(this.API_URI+'/alumno/'+alumno.id, alumno);
  }

  deleteAlumno(id: number){
    return this.http.delete(this.API_URI+'/alumno/'+id);
  }
}
